### PR TITLE
Add pygeos recipe

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example) -> going to create license first in the repo (https://github.com/caspervdw/pygeos/pull/39)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@caspervdw 

cc @ocefpaf

### CODE REVIEW

File: recipes/pygeos/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy
+    - geos
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - geos
```
Comment (jorisvandenbossche) [inline recipes/pygeos/meta.yaml]: @ocefpaf does `geos` need to be listed here? (eg shapely does not do it, although it is a run dependency as well?)

Comment (ocefpaf) [reply recipes/pygeos/meta.yaml]: geos has run_exports, so you don't have to list but it does not hurt to have it there.

Comment (jorisvandenbossche) [reply recipes/pygeos/meta.yaml]: Thanks. And for geos it does not need to be wrapped in a pin_compatible?

Comment (ocefpaf) [reply recipes/pygeos/meta.yaml]: No need. The global pinning + run_exports should take care of that. Numpy is a special case.

**Comment by ocefpaf** [comment]
I'll try to fix Windows tomorrow @jorisvandenbossche.

**Comment by jorisvandenbossche** [comment]
Thanks for the other patches! (I was thinking to maybe just wait on the next release (assuming that indeed all those things already have been fixed on master), but it seems that you fixed it with a minimal patch here as well now)

**Comment by ocefpaf** [comment]
> Thanks for the other patches! (I was thinking to maybe just wait on the next release

Works for me. But I believe we are close to a working package here.

> (assuming that indeed all those things already have been fixed on master), but it seems that you fixed it with a minimal patch here as well now)

They are. We can test it by building it from master and waiting for the release but the new `setup.py` upstream seems to cover all these issues.

**Comment by jorisvandenbossche** [comment]
If we can get it working with the current patches, I would prefer to go forward. Then there is already a feedstock and updating it once the next release is out is easier.

**Comment by ocefpaf** [comment]
> If we can get it working with the current patches, I would prefer to go forward. Then there is already a feedstock and updating it once the next release is out is easier.

Agreed. We can also test the latest master in the feedstock just to be sure.

**Comment by ocefpaf** [comment]
@jorisvandenbossche seems that the errors now are real and I'm thinking about skipping Windows for now to get this merged.

**Comment by jorisvandenbossche** [comment]
Skipping windows to get an initial feedstock is certainly fine for me

**Comment by jorisvandenbossche** [comment]
@ocefpaf thanks a lot!

**Pull request merged**